velox/functions/prestosql/JsonFunctions.h

```diff
@@ -74,4 +74,36 @@ struct JsonArrayLengthFunction {
   }
 };
 
+template <typename T>
+struct JsonArrayContainsFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  template <typename TInput>
+  FOLLY_ALWAYS_INLINE bool
+  call(bool& result, const arg_type<Varchar>& json, const TInput& value) {
+    auto parsedJson = folly::parseJson(json);
+    if (!parsedJson.isArray()) {
+      return false;
+    }
+
+    result = false;
+    for (const auto& v : parsedJson) {
+      if (std::is_same_v<TInput, bool> && v.isBool() && v == value) {
```

Comment:
The checks std::is_same_v<TInput, bool> for TInput can be done only once outside the loop. 
It is inefficient to do for each value in the json array.

So outside the loop you have valueBool = std::is_same_v<TInput, bool>;

and inside the loop
if (valueBool && v.isBool() && v == value) {...}

Reply:
Thanks for the suggestion, changed accordingly